### PR TITLE
HangWatcher to nightly 50%

### PR DESCRIPTION
**Pull request by atuchin-m**
Nightly 50%, tested manually.
Security review: brave/security#659

### What the study does?
* enables sending crash dumps if a hang happens. The logic of hang detections are chromium and already works everywhere, but record only UMA.

### What can go wrong? Is data loss possible?
* A lot of dumps because a lot of hangs or false positives.

`min_version` is set to current nightly with landed https://github.com/brave/brave-browser/issues/19822
(the changed description in the settings page).

PRs to master:
https://github.com/brave/brave-variations/pull/192
https://github.com/brave/brave-variations/pull/193

The root issue:
https://github.com/brave/brave-browser/issues/20304

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following builds:

```
Brave | 1.36.22 Chromium: 97.0.4692.71 (Official Build) nightly (64-bit)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 11 Version 21H2 (Build 22000.376)
```

```
Brave | 1.35.75 Chromium: 97.0.4692.71 (Official Build) beta (64-bit)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 11 Version 21H2 (Build 22000.376)
```

```
Brave | 1.34.80 Chromium: 97.0.4692.71 (Official Build) (64-bit)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 11 Version 21H2 (Build 22000.376)
```

STR/Cases:

* ensured that both `HangWatcher:HangWatcherEnableDumps` & `HangWatcher:Default` are being used under Nightly
* ensured that both `HangWatcher:HangWatcherEnableDumps` & `HangWatcher:Default` are not being used under `Beta` & `Release`

**`Nightly Enabled`** | **`Nightly Disabled`**
----------------
NightlyEnabled
--------|--------------------------

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` for `Nightly` at `50%` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-variations/pull/194#issuecomment-1007705716.

**Pull request merged**
